spec/registry.md

```diff
@@ -0,0 +1,156 @@
+# WIP DRAFT MessageFormat 2.0 Registry
+
+_This document is non-normative._
+
+The implementations and tooling can greatly benefit from a structured definition of formatting and matching functions available to messages at runtime.
+The _registry_ is a mechanism for storing such declarations in a portable manner.
+
+## Goals
+
+The goal of the registry is to provide a machine-readable description of MessageFormat extensions (custom functions) in order to:
+
+* support the localization roundtrip,
+* validate message correctness, and
+* improve the authoring experience.
+
+## Use Cases
+
+The registry enables the following usage scenarios:
+
+* Generate variant keys for a given locale during XLIFF extraction.
+* Verify the exhaustiveness of variant keys given a selector.
+* Type-check values passed into functions.
+* Validate that matching functions are only called in selectors.
```

Comment:
You're right, let's be explicit about this. I'd prefer to leave this out of this PR, however, because I think that we will first need to decide whether it's allowed to match on a function-less expression (i.e. `match {$foo}` instead of `match {$foo :func}`).

Reply:
Filed #409 to discuss this further.